src/ranged.cpp

```diff
@@ -1177,13 +1182,58 @@ dealt_projectile_attack player::throw_item( const tripoint &target, const item &
     return dealt_attack;
 }
 
+void practice_archery_proficiency( player &p, const item &relevant )
+{
+    // Do nothing, we are not doing archery
+    if( relevant.gun_skill() != skill_archery ) {
+        return;
+    }
+
+    // We are already a master, practice has no effect
+    if( p.has_proficiency( proficiency_prof_bow_master ) ) {
+        return;
+    }
+
+    // Only train archery in multiples of 100 moves (or 1 turn) for performance, focus, and simplifying calculations
+    if( p.archery_aim_counter++ > 100 ) {
```

Comment:
As is, this will check if `p.archery_aim_counter` is greater than 100, add 1 to it, then set it to 0.
The correct way to do this would be
```suggestion
    if( ++p.archery_aim_counter > 100 ) {
```
But, I think this style is confusing and uncessary, and would prefer it it were like this:
```suggestion
    p.archery_aim_counter++;
    if( p.archery_aim_counter > 100 ) {
```
There's some talk about `++a` being more efficient than `a++` (because a++ returns a copy of the value before 1 is added to it), and my personal style is to always do `++a`, but it's a micro optimization and you shouldn't care about that unless you want to (or it's in statements where it makes a functional difference, like this one).